### PR TITLE
Updated: JRuby to 9.2.11.0

### PR DESCRIPTION
**Pull request by kares**
https://repo.maven.apache.org/maven2/org/jruby/jruby-dist/9.2.11.0/jruby-dist-9.2.11.0-bin.tar.gz.sha1

### CODE REVIEW

File: rubyUtils.gradle [hunk 1]
```diff
@@ -81,7 +81,7 @@ void gem(File projectDir, File buildDir, String gem, String version, String path
         jruby.runScriptlet("""
                 require 'rubygems/commands/install_command'
                 cmd = Gem::Commands::InstallCommand.new
-                cmd.handle_options ['--no-ri', '--no-rdoc', '${gem}', '-v', '${version}', '-i', '${path}']
+                cmd.handle_options ['--no-document', '${gem}', '-v', '${version}', '-i', '${path}']
```
Comment (jsvd) [inline rubyUtils.gradle]: the end of an era :)

**Comment by kares** [comment]
so this is now :green_salad: ... initially thought there's a JRuby regression but turns out LS was doing `RubyString#match` using the native API - which [requires a frame](https://gist.github.com/kares/85669d9834e25889b4e99075b7b22308) to be present (in order to store `$~`)

have refactored to using `match?` (does not backref and friends) which makes sense as captures are not being used.

File: rubyUtils.gradle (hunk 1 above)
Comment (kares) [reply rubyUtils.gradle]: `--no-rdoc` is dead long live `no-document` :new_moon_with_face:

**Comment by kares** [comment]
extracted the light-weight matching part into: https://github.com/elastic/logstash/pull/11653
since that potentially makes sense as a bugfix even on versions that are stuck with < 9.2.11

**Pull request closed without merging**